File: app/src/main/java/digispark/tech/ganapatiaarti/Splash_Screen.java
package digispark.tech.ganapatiaarti;

import android.Manifest;
import android.app.Activity;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;

import androidx.annotation.NonNull;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import java.util.ArrayList;
import java.util.List;

import digispark.tech.ganapatiaarti.utils.UserInterfaceUtils;


public class Splash_Screen extends Activity {

    Handler handler = new Handler();
    private final int MULTIPLE_PERMISSIONS = 10;
    private String[] permissions = {Manifest.permission.INTERNET,
            Manifest.permission.ACCESS_NETWORK_STATE,
            Manifest.permission.WAKE_LOCK,
            Manifest.permission.ACCESS_COARSE_LOCATION,
            Manifest.permission.ACCESS_FINE_LOCATION};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash__screen);
    }

    @Override
    protected void onStart() {
        super.onStart();
        handler.postDelayed(r, 2000);
    }

    Runnable r = new Runnable() {
        public void run() {
            if (checkPermissions()) {
                //  permissions  granted.
                determineNavigation();
            }
        }
    };

    private void determineNavigation(){
        startActivity(new Intent(this, MainActivity.class));
        this.finish();
    }

    private boolean checkPermissions(){
        int result;
        List<String> listPermissionsNeeded = new ArrayList<String>();
        for (String p : permissions){
            result = ContextCompat.checkSelfPermission(this, p);
            if (result != PackageManager.PERMISSION_GRANTED)
                listPermissionsNeeded.add(p);
        }
        if (!listPermissionsNeeded.isEmpty()){
            ActivityCompat.requestPermissions(this, listPermissionsNeeded.toArray(new String[0]), MULTIPLE_PERMISSIONS);
            return false;
        }
        return true;
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        switch (requestCode){
            case MULTIPLE_PERMISSIONS:
                boolean permissionDenied = false;
                for (int granted : grantResults) {
                    if (granted == PackageManager.PERMISSION_DENIED) {
                        permissionDenied = true;
                        UserInterfaceUtils.showToast(this, getString(R.string.please_grant_all_permissions));
                        break;
                    }
                }
                if (!permissionDenied)
                    startActivity(new Intent(this, MainActivity.class));
                this.finish();

                break;
        }
    }

    @Override
    public void onBackPressed() {

        super.onBackPressed();
        try {
            finish();
            if (handler != null) {
                handler.removeMessages(0);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

}

File: app/src/main/java/digispark/tech/ganapatiaarti/Splash_Screen.kt
package digispark.tech.ganapatiaarti

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.Handler
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.calibehr.mitra.utils.SharedPreferencesHelper
import digispark.tech.ganapatiaarti.base.BaseActivity
import digispark.tech.ganapatiaarti.constants.Constant

import java.util.ArrayList

import digispark.tech.ganapatiaarti.utils.UserInterfaceUtils


class Splash_Screen : Activity() {

    internal var handler: Handler? = Handler()
    private val MULTIPLE_PERMISSIONS = 10
    private val permissions = arrayOf(Manifest.permission.INTERNET,
            Manifest.permission.ACCESS_NETWORK_STATE, Manifest.permission.WAKE_LOCK,
            Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION)

    internal var r: Runnable = Runnable {
        if (checkPermissions()) {
            //  permissions  granted.
            determineNavigation()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash__screen)

        val sp = SharedPreferencesHelper.getPref(this@Splash_Screen, Constant.SP_LANG_CODE)
        if (sp!!.isNotEmpty()){
            Constant.LANG_CODE = sp
        }
    }

    override fun onStart() {
        super.onStart()
        handler?.postDelayed(r, 2000)
    }

    private fun determineNavigation() {
        startActivity(Intent(this, MainActivity::class.java))
        this.finish()
    }

    private fun checkPermissions(): Boolean {
        var result: Int
        val listPermissionsNeeded = ArrayList<String>()
        for (p in permissions) {
            result = ContextCompat.checkSelfPermission(this, p)
            if (result != PackageManager.PERMISSION_GRANTED)
                listPermissionsNeeded.add(p)
        }
        if (!listPermissionsNeeded.isEmpty()) {
            ActivityCompat.requestPermissions(this, listPermissionsNeeded.toTypedArray<String>(), MULTIPLE_PERMISSIONS)
            return false
        }
        return true
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
        when (requestCode) {
            MULTIPLE_PERMISSIONS -> {
                var permissionDenied = false
                for (granted in grantResults) {
                    if (granted == PackageManager.PERMISSION_DENIED) {
                        permissionDenied = true
                        UserInterfaceUtils.showToast(this, getString(R.string.please_grant_all_permissions))
                        break
                    }
                }
                if (!permissionDenied)
                    startActivity(Intent(this, MainActivity::class.java))
                this.finish()
            }
        }
    }

    override fun onBackPressed() {

        super.onBackPressed()
        try {
            finish()
            if (handler != null) {
                handler?.removeMessages(0)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onDestroy() {
        super.onDestroy()
    }

}
